api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -104,32 +105,32 @@
       []
       reducible)))
 
-(defn after-forms-loaded [{::keys [forms-instances-grouped-by-form
-                                   unilog-id
-                                   form-updated
-                                   form-deleted
-                                   form-instance-deleted
-                                   data-point-changed
-                                   data-point-deleted
-                                   survey-changed
-                                   survey-deleted]}
-                          form-id->form]
+(defn filter-events-by-authorization [{::keys [forms-instances-grouped-by-form
+                                               unilog-id
+                                               form-updated
+                                               form-deleted
+                                               form-instance-deleted
+                                               data-point-changed
+                                               data-point-deleted
+                                               survey-changed
+                                               survey-deleted]}
+                                      form-id->form
+                                      authorized-forms
+                                      authorized-surveys]
   {:unilog-id unilog-id
-   :form-changed (->> form-updated
-                   (keep form-id->form)
-                   set)
-   :form-deleted form-deleted
+   :form-changed (->> form-updated (keep form-id->form) set)
+   :form-deleted (set/intersection authorized-forms form-deleted)
    :form-instance-deleted form-instance-deleted
    :form-instances-to-load (->> forms-instances-grouped-by-form
-                             (keep (fn [[form-id form-instance]]
-                                     (when-let [form (get form-id->form form-id)]
-                                       {:form form
-                                        :form-instance-ids (set (map :id form-instance))})))
-                             set)
-   :data-point-changed data-point-changed
+                                (keep (fn [[form-id form-instance]]
+                                        (when-let [form (get form-id->form form-id)]
+                                          {:form form
+                                           :form-instance-ids (set (map :id form-instance))})))
+                                set) ;; the mapping form-id->form filter instances
```

Comment:
This comment sounds like the code is not clear. Please fix the code and remove the comment